pkg/podstate/pod_state_test.go

```diff
@@ -0,0 +1,167 @@
+			expectedList: []framework.NodeScore{{Name: "node1", Score: framework.MinNodeScore}, {Name: "node2", Score: framework.MaxNodeScore}},
+			name:         "node has less (terminatingPodNumber - nominatedPodNumber) get lower score",
+		},
+		{
+			nodeInfos:    []*framework.NodeInfo{makeNodeInfo("node1", 5, 0), makeNodeInfo("node2", 3, 1), makeNodeInfo("node3", 2, 1), makeNodeInfo("node4", 0, 1)},
+			expectedList: []framework.NodeScore{{Name: "node1", Score: framework.MaxNodeScore}, {Name: "node2", Score: 50}, {Name: "node3", Score: 33}, {Name: "node4", Score: framework.MinNodeScore}},
+			name:         "node has more (terminatingPodNumber - nominatedPodNumber) get higher score",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cs := clientsetfake.NewSimpleClientset()
+			informerFactory := informers.NewSharedInformerFactory(cs, 0)
+			registeredPlugins := []st.RegisterPluginFunc{
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+				st.RegisterPluginAsExtensions(Name, New, "Score"),
+			}
+			fakeSharedLister := &fakeSharedLister{nodes: test.nodeInfos}
+
+			fh, err := st.NewFramework(
+				registeredPlugins,
+				frameworkruntime.WithClientSet(cs),
+				frameworkruntime.WithInformerFactory(informerFactory),
+				frameworkruntime.WithSnapshotSharedLister(fakeSharedLister),
+				frameworkruntime.WithPodNominator(testutil.NewPodNominator()),
+			)
+			if err != nil {
+				t.Fatalf("fail to create framework: %s", err)
+			}
+			// initialize nominated pod by adding nominated pods into nominatedPodMap
+			for _, n := range test.nodeInfos {
+				for _, p := range n.Pods {
+					if p.Pod.Status.NominatedNodeName != "" {
+						addNominatedPod(p.Pod, n.Node().Name, fh.PreemptHandle())
+					}
+				}
+			}
+			pe, _ := New(nil, fh)
+			var gotList framework.NodeScoreList
+			plugin := pe.(framework.ScorePlugin)
+			for i, n := range test.nodeInfos {
+				score, err := plugin.Score(context.Background(), nil, nil, n.Node().Name)
+				if err != nil {
+					t.Errorf("unexpected error: %v", err)
+				}
+				gotList = append(gotList, framework.NodeScore{Name: test.nodeInfos[i].Node().Name, Score: score})
+			}
+
+			status := plugin.ScoreExtensions().NormalizeScore(context.Background(), nil, nil, gotList)
+			if !status.IsSuccess() {
+				t.Errorf("unexpected error: %v", status)
+			}
+
+			for i := range gotList {
+				if test.expectedList[i].Score != gotList[i].Score {
+					t.Errorf("expected %#v, got %#v", test.expectedList[i].Score, gotList[i].Score)
+				}
+			}
+		})
+	}
+}
+
+func makeNodeInfo(node string, terminatingPodNumber, nominatedPodNumber int) *framework.NodeInfo {
+	ni := framework.NewNodeInfo()
+	for i := 0; i < terminatingPodNumber; i++ {
+		podInfo := &framework.PodInfo{
+			Pod: makeTerminatingPod(fmt.Sprintf("tpod_%s_%s", node, strconv.Itoa(i+1))),
```

Comment:
We don't need `strconv.Itoa`, do we?

```suggestion
			Pod: makeTerminatingPod(fmt.Sprintf("tpod_%s_%v", node, i+1)),
```